easybuild/tools/build_from_specsfile.py

```diff
@@ -0,0 +1,150 @@
+        return easyconfigs_full_paths
+
+    # single command
+    def get_ec_path(self, sw):
+        full_path = search_easyconfigs(query='%s-%s-%s-%s' % (
+            str(sw.software), str(sw.version), str(sw.toolchain_name), str(sw.toolchain_version)),
+            short=False, filename_only=False,
+            terse=False, consider_extra_paths=True, print_result=False, case_sensitive=False)
+        if len(full_path) == 1:
+            return full_path[0]
+        else:
+            print('%s does not have clearly specified parameters - %s matches found. Skipping. \n' %
+                  (sw.software, str(len(full_path))))
+
+    # todo: flags applicable to all sw (i.e. robot)
+    def get_general_options(self):
+        general_options = {}
+        general_options['robot'] = self.robot
+        general_options['easybuild_version'] = self.easybuild_version
+        return general_options
+
+
+# single sw command
+class SoftwareSpecs(object):
+    def __init__(self, software, version, toolchain, toolchain_version, toolchain_name):
+        self.software = software
+        self.version = version
+        self.toolchain = toolchain
+        self.toolchain_version = toolchain_version
+        self.toolchain_name = toolchain_name
+        self.toolchain = toolchain
+
+        self.version_suffix = None
+
+    def get_version_suffix(self):
+        return self.version_suffix or ''
+
+
+# implement this to your own needs - to create custom yaml/json/xml parser
+class GenericSpecsParser(object):
+    @staticmethod
+    def parse(filename):
+        raise NotImplementedError
+
+
+class YamlSpecParser(GenericSpecsParser):
+    @staticmethod
+    def parse(filename):
+
+        try:
+            with open(filename, 'r') as f:
+                spec_dict = yaml.safe_load(f)
+
+            eb = Specsfile()
+        except FileNotFoundError:
+            raise EasyBuildError("Could not read provided specsfile.")
+
+        sw_dict = spec_dict["software"]
+
+        # assign software-specific EB attributes
+        for software in sw_dict:
+            try:
+                # iterates through toolchains to find out what sw version is needed
+                for yaml_toolchain in sw_dict[software]['toolchains']:
+                    # retrieves version number
+                    for yaml_version in sw_dict[software]['toolchains'][yaml_toolchain]['versions']:
+
+                        # if among versions there is anything else than known flags or version flag itself, it is wrong
+                        # identification of version strings is vague
+                        if str(yaml_version)[0].isdigit() \
+                                or str(yaml_version)[-1].isdigit():
+                            # creates a sw class instance
+                            try:
+                                yaml_toolchain_name = str(yaml_toolchain).split('-', 1)[0]
+                                yaml_toolchain_version = str(yaml_toolchain).split('-', 1)[1]
```

Comment:
This can be simplified a lot I think...
```python
if '-' in toolchain:
    toolchain_name, toolchain_version = toolchain.split('-', 1)
else:
    toolchain_name, toolchain_version = toolchain, ''
```

No more need for `try`/`except` in that case?